From ab60659ba72f1da318881d48fe1a4fc3ba737a9a Mon Sep 17 00:00:00 2001
From: Anurag Kumar Vulisha <anurag.kumar.vulisha@xilinx.com>
Date: Thu, 23 Jan 2020 19:51:00 +0530
Subject: [PATCH 1011/1851] usb: dwc3: Fix missing peripheral irq error message

commit 8c4a2a2ee6415232edbd34c0cfc5ecbc5d242f78 from
https://github.com/Xilinx/linux-xlnx.git

Due to incorrect programming sequence the "missing peripheral IRQ"
error message gets showed even after finding the correct IRQ.
This patch fixes the code logic for removing this error message

Signed-off-by: Anurag Kumar Vulisha <anurag.kumar.vulisha@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/dwc3/gadget.c | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 7bac82be544e..95d59e037029 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -3442,23 +3442,26 @@ static int dwc3_gadget_get_irq(struct dwc3 *dwc)
 	if (irq > 0)
 		dwc->irq_gadget = irq;
 
+	if (irq == -EPROBE_DEFER)
+		goto out;
+
 	/* look for wakeup interrupt if hibernation is supported */
 	if (dwc->has_hibernation) {
 		irq = platform_get_irq(dwc3_pdev, 2);
-		if (irq <= 0) {
-			if (irq != -EPROBE_DEFER)
-				dev_err(dwc->dev, "missing wakeup IRQ\n");
-			if (!irq)
-				irq = -EINVAL;
-			return irq;
-		}
+		if (irq > 0)
+			dwc->irq_wakeup = irq;
 
-		dwc->irq_wakeup = irq;
+		if (irq == -EPROBE_DEFER)
+			goto out;
 	}
 
-	if (!irq)
-		irq = -EINVAL;
+	if (irq <= 0) {
+		if (irq != -EPROBE_DEFER)
+			dev_err(dwc->dev, "missing peripheral IRQ\n");
 
+		if (!irq)
+			irq = -EINVAL;
+	}
 out:
 	return irq;
 }
-- 
2.31.1

